.github/workflows/branches-and-prs.main.kts

```diff
@@ -0,0 +1,173 @@
+        group = "${expr { github.workflow }}-${expr("${Contexts.github.eventPullRequest.pull_request.number} || ${Contexts.github.ref}")}",
+        cancelInProgress = true
+    )
+) {
+    job(
+        id = "check_all_workflow_yaml_consistency",
+        name = "Check all Workflow YAML consistency",
+        runsOn = UbuntuLatest
+    ) {
+        uses(
+            name = "Checkout Repository",
+            action = CheckoutV3()
+        )
+        run(
+            name = "Regenerate all workflow YAMLs and check for modifications",
+            command = """find .github/workflows -mindepth 1 -maxdepth 1 -name "*.main.kts" | xargs -ri sh -c '{} && git diff --exit-code'"""
+        )
+    }
+
+    val (javaVersions, variants) = getMatrixAxes()
+    job(
+        id = "build-and-verify",
+        name = "Build and Verify",
+        runsOn = RunnerType.Custom(expr("matrix.os")),
+        _customArguments = mapOf(
+            "strategy" to mapOf(
+                "fail-fast" to false,
+                "matrix" to mapOf(
+                    "os" to listOf("ubuntu-latest"),
+                    "variant" to variants,
+                    "java" to javaVersions,
+                    "exclude" to javaVersions
+                        .filter { it.toInt() >= 17 }
+                        .map { javaVersion ->
+                            mapOf(
+                                "os" to "ubuntu-latest",
+                                "variant" to "2.5",
+                                "java" to javaVersion
+                            )
+                        },
+                    "include" to listOf("windows-latest", "macos-latest")
+                        .flatMap { os -> variants.map { os to it } }
+                        .map { (os, variant) ->
+                            mapOf(
+                                "os" to os,
+                                "variant" to variant,
+                                "java" to javaVersions.first()
+                            )
+                        }
+                )
+            )
+        )
+    ) {
+        uses(
+            name = "Checkout Repository",
+            action = CheckoutV3(
+                // Codecov needs fetch-depth > 1
+                fetchDepth = CheckoutV3.FetchDepth.Value(2)
+            )
+        )
+        uses(
+            name = "Set up JDKs",
+            action = SetupBuildEnv(
+                additionalJavaVersion = expr("matrix.java")
+            )
+        )
+        val SPOCK_BUILD_CACHE_USERNAME by Contexts.secrets
+        val SPOCK_BUILD_CACHE_PASSWORD by Contexts.secrets
+        val GRADLE_ENTERPRISE_ACCESS_KEY by Contexts.secrets
+        uses(
+            name = "Build Spock",
+            action = GradleBuildActionV2(
+                arguments = listOf(
+                    "--no-parallel",
+                    "--stacktrace",
+                    "ghActionsBuild",
+                    """"-Dvariant=${expr("matrix.variant")}"""",
```

Comment:
In this specific case, it indeed is mainly preference between
```kts
""""-Dvariant=${expr(Matrix.variant)}""""
```
and
```kts
""""-Dvariant=${'$'}{{ ${Matrix.variant} }}""""
```
and
```kts
"\"-Dvariant=\${{ ${Matrix.variant} }}\""
```
while when doing it manually, you also have to make sure not to forget the mandatory spaces within the curly braces.

Besides that it is also more consistent with the `expr { ... }` cases.

Reply:
Wouldn't it be `'-Dvariant=${{ matrix.variant }}'` as there is no groovy templating, but ok.

Reply:
This is Kotlin, not Groovy. ;-)
And even if this were possible - which it is not - it would still bear the unnecessary risk to forget the spaces or otherwise mess-up the syntax and make it less consistent. :-)